packages/gatsby/src/schema/schema.js

```diff
@@ -741,8 +741,10 @@ const addTypeToRootQuery = ({ schemaComposer, typeComposer }) => {
   const queryName = _.camelCase(typeName)
   const queryNamePlural = _.camelCase(`all ${typeName}`)
   schemaComposer.Query.addFields({
-    [queryName]: typeComposer.getResolver(`findOne`),
-    [queryNamePlural]: typeComposer.getResolver(`findManyPaginated`),
+    [queryName]: new GraphQLNonNull(typeComposer.getResolver(`findOne`)),
```

Comment:
This technically isn't guaranteed to always not be null

Reply:
@pieh What would a scenario were that field returns `null` look like?

Can a page still be successfully built or does it indicate an error that fails the build?

Reply:
It might not be reasonable, but you can do
```
someType(id: { eq: "id-of-node-that-doesnt-exist" }) {
  id
}
```

Which will return `null` because node can't be found.

Other scenarios are - with schema customizations you can define types even if there are no nodes for that type - so those will always be empty (even without any filters)

Reply:
I do get the first scenario. Marking the field as NonNull could still be better, as it would uncover an error right away. My assumption is that you always require the data you want to find.

If there are fields where you try to find something dynamically **and** don't know about it's existence **and** handle the case of not finding something, a nullable field could make sense.

In the case of the `allSomeType` fields - i think those should always return at least an empty list, so `[SomeType]!` instead of `[SomeType]`.